process/block/metablock.go

```diff
@@ -587,7 +566,7 @@ func (mp *metaProcessor) RestoreBlockIntoPools(headerHandler data.HeaderHandler,
 			continue
 		}
 
-		headerPool.AddHeader(hdrHash, shardHeader)
+		mp.dataPool.Headers().AddHeader(hdrHash, shardHeader)
```

Comment:
take the mp.dataPool.Headers into a variable.

Reply:
Done